k8s/sidecar/sidecar_webhook.go

```diff
@@ -80,204 +86,107 @@ func (r *SidecarInjector) Handle(ctx context.Context, req admission.Request) adm
 		sidecar.Name = selectorname + "-default-sidecar"
 		sidecar.Namespace = pod.Namespace
 
-		err := r.Client.Get(ctx, types.NamespacedName{Name: sidecar.Name, Namespace: sidecar.Namespace}, &sidecar)
+		err := r.Client.Get(ctx, types.NamespacedName{Name: sidecar.Name, Namespace: sidecar.Namespace}, sidecar)
 		if err != nil && !apierrors.IsNotFound(err) {
 			logger.Info("Get sidecar cr failed", "error", err)
 			return admission.Errored(http.StatusInternalServerError, err)
 		}
 		// if the default sidecar cr doesn't exist, create it
 		if apierrors.IsNotFound(err) {
 			sidecar.Spec.Replicas = 1
+			// use default configurations
 			sidecar.Spec.Selector = keys[0] + "=" + labels[keys[0]]
-			if err := r.Client.Create(ctx, &sidecar); err != nil {
+			sidecar.Spec.VineyardConfig.EtcdEndpoint = "http://etcd-for-vineyard:2379"
+			sidecar.Spec.VineyardConfig.EtcdPrefix = "/vineyard"
+			sidecar.Spec.VineyardConfig.Image = "vineyardcloudnative/vineyardd:latest"
+			sidecar.Spec.VineyardConfig.ImagePullPolicy = "IfNotPresent"
+			sidecar.Spec.VineyardConfig.Socket = "/var/run/vineyard.sock"
+			sidecar.Spec.VineyardConfig.Size = "256Mi"
+			sidecar.Spec.VineyardConfig.StreamThreshold = 80
```

Comment:
Why do we need these hard-coded things here?

I think they should be specified in the sidecar's yaml?

Reply:
Can that be fixed? It not, maybe moving those constants to `_types.go` is a better solution.

Reply:
Fixed. It's actually a bug, as the default value is not the same between the nested structure and its' sub-fields.